### PR TITLE
builds create prompt ui and api

### PR DESCRIPTION
**Pull request by H-b8**
Builds UI for `pages/prompt/new.js`.

### CODE REVIEW

**Review by MilcahHalili — CHANGES_REQUESTED** [review]
This is a great start, @H-b8! Can you also build out the api route so that, on click of the submit button, a new prompt is created in the database?

**Comment by H-b8** [comment]
- Built off of the `api/prompt/create` you started and have a successful post to db. Test @ /prompt/new
- Refactored code in a few places to get rid of warnings in the console
- Style adjustments
- Refactored Prompt component to show author on index page
- Refactored `prompt/[id]` page

File: components/Prompt.js [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
   })
 
   return (
-    <div onClick={() => Router.push("/prompt/[id]", `/prompt/${prompt.id}`)} className={Styles.prompt}>
-      <h2 className={Styles.prompth2}>{prompt.title}</h2> 
-      <p className={Styles.instructorName}>By {instructorName}</p>
-      <ReactMarkdown source={prompt.content} />
+    <div onClick={() => handleClick()} className={Styles.prompt}>
+      <h2 className={Styles.prompth2}>{prompt.title}</h2>
+      <p className={Styles.instructorName}>By {authorName || 'Unknown Instrcutor'}</p>
```
Comment (MilcahHalili) [inline components/Prompt.js:29]: Typo in `Instructor`!

File: pages/api/prompt/create.js [hunk 2]
```diff
@@ -5,14 +5,19 @@ export default function handler(req, res) {
...
     async function main() {
+      const user = await prisma.user.findUnique({
+        where: {
+          email: req.body.email
+        }
+      });
       const prompt = await prisma.prompt.create({
         data: {
-          instructorId: 1,
+          authorId: user.id,
```
Comment (MilcahHalili) [inline pages/api/prompt/create.js:15]: Thanks for removing the hard-coded value/making this dynamic!

File: pages/prompt/[id].js [hunk 3]
```diff
@@ -14,31 +13,18 @@ const QuillNoSSRWrapper = dynamic(import('react-quill'), {
...
-        .then(res => res.json())
-        .then(res => {
-          setData(res)
-        })
-        .catch(err => console.error(err))
-    }
-  }
+    const promptData = await fetch(`../api/prompt/${router.query.id}`);
+    const res = await promptData.json();
+    setData(res);
```
Comment (MilcahHalili) [inline pages/prompt/[id].js:26]: Ooo this is so much cleaner, tysm!

File: pages/prompt/[id].js [hunk 4]
```diff
@@ -58,15 +44,16 @@ const Prompt = () => {
...
 
   const handleSubmit = async e => {
-    await e.preventDefault()
-    await createPost(data)
-    console.log('yes!')
-    getOnePrompt()
+    await e.preventDefault();
+    await createPost();
+    setPostContent('');
+    console.log('yes!');
```
Comment (MilcahHalili) [inline pages/prompt/[id].js:50]: Lol we can remove this console log now, thanks!

File: pages/prompt/[id].js [hunk 5]
```diff
@@ -1,5 +1,4 @@
 import React, { useEffect, useState } from "react"
-import ReactMarkdown from "react-markdown"
```
Comment (MilcahHalili) [inline pages/prompt/[id].js:2]: Nice!

**Review by MilcahHalili — CHANGES_REQUESTED** [review]
Amazing! Excellent work, @H-b8. Just a couple tiny change requests, and then we'll be good to go.

**Comment by H-b8** [comment]
Good eye @MilcahHalili ! Console removed and typo fixed 😄

**Review by MilcahHalili — APPROVED** [review]
Well done!

**Pull request merged**